pkg/controllers/membercluster/v1beta1/membercluster_controller_integration_test.go

```diff
@@ -842,4 +855,215 @@ var _ = Describe("Test MemberCluster Controller", func() {
+
+			By("trigger reconcile again to update member cluster status to joined")
+			result, err = r.Reconcile(ctx, ctrl.Request{
+				NamespacedName: memberClusterNamespacedName,
+			})
+			Expect(result).Should(Equal(ctrl.Result{}))
+			Expect(err).Should(Succeed())
+
+			By("getting imc status")
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: memberClusterName, Namespace: namespaceName}, &imc)).Should(Succeed())
+
+			By("checking mc status")
+			Expect(k8sClient.Get(ctx, memberClusterNamespacedName, mc)).Should(Succeed())
+
+			wantMC := clusterv1beta1.MemberClusterStatus{
+				Conditions: []metav1.Condition{
+					{
+						Type:               string(clusterv1beta1.ConditionTypeMemberClusterReadyToJoin),
+						Status:             metav1.ConditionTrue,
+						Reason:             reasonMemberClusterReadyToJoin,
+						ObservedGeneration: mc.GetGeneration(),
+					},
+					{
+						Type:               string(clusterv1beta1.ConditionTypeMemberClusterJoined),
+						Status:             metav1.ConditionTrue,
+						Reason:             reasonMemberClusterJoined,
+						ObservedGeneration: mc.GetGeneration(),
+					},
+					{
+						Type:               propertyProviderConditionType1,
+						Status:             propertyProviderConditionStatus1,
+						Reason:             propertyProviderConditionReason1,
+						Message:            propertyProviderConditionMessage1,
+						ObservedGeneration: mc.GetGeneration(),
+					},
+					{
+						Type:               propertyProviderConditionType2,
+						Status:             propertyProviderConditionStatus2,
+						Reason:             propertyProviderConditionReason2,
+						Message:            propertyProviderConditionMessage2,
+						ObservedGeneration: mc.GetGeneration(),
+					},
+				},
+				Properties:    imc.Status.Properties,
+				ResourceUsage: imc.Status.ResourceUsage,
+				AgentStatus:   imc.Status.AgentStatus,
+			}
+			Expect(cmp.Diff(wantMC, mc.Status, ignoreOption)).Should(BeEmpty())
+		})
+
+		AfterEach(func() {
+			var ns corev1.Namespace
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: namespaceName}, &ns)).Should(Succeed())
+			By("Deleting the namespace")
+			Eventually(func() error {
+				return k8sClient.Delete(ctx, &ns)
+			}, timeout, interval).Should(SatisfyAny(Succeed(), &utils.NotFoundMatcher{}))
+		})
+
+		It("force delete where member agent never updates Internal Member Cluster", func() {
+			By("simulate delete member cluster")
+			var mc clusterv1beta1.MemberCluster
+			Expect(k8sClient.Get(ctx, memberClusterNamespacedName, &mc)).Should(Succeed())
+			Expect(k8sClient.Delete(ctx, &mc)).Should(Succeed())
+
+			By("trigger reconcile to initiate leave workflow")
+			result, err := r.Reconcile(ctx, ctrl.Request{
+				NamespacedName: memberClusterNamespacedName,
+			})
+			Expect(result).Should(Equal(ctrl.Result{RequeueAfter: r.ForceDeleteWaitTime}))
+			Expect(err).Should(Succeed())
+
+			By("wait until force delete wait time has crossed")
```

Comment:
Just a fancy way to sleep, please lmk if there is a better way to do this

Reply:
why wait before the check? The expected behavior is the namespace should be deleting after the delete wait time passes so the wait should be after.

Reply:
That can be done for E2Es but since it's an we need to manually trigger the reconcile to simulate the Requeue After the force delete wait time